src/Nethermind/Nethermind.Runner/Ethereum/Steps/StartRpc.cs

```diff
@@ -105,5 +114,28 @@ await jsonRpcRunner.Start(cancellationToken).ContinueWith(x =>
                 if (logger.IsInfo) logger.Info("Json RPC is disabled");
             }
         }
+        private void ConfigureJwtSecret(IKeyStoreConfig keyStoreConfig, ILogger logger, IJsonRpcConfig jsonRpcConfig) {
+
+            string newPath = Path.GetFullPath(Path.Join(keyStoreConfig.KeyStoreDirectory, "jwt-secret"));
+            string oldPath = Path.GetFullPath("keystore/jwt-secret");
+            // check if jwt-secret file already exists in previous default directory
+            if (!File.Exists(newPath) && File.Exists(oldPath))
+            {
+                try
+                {
+                    File.Move(oldPath, newPath);
+                    if (logger.IsWarn) logger.Warn($"Moved JWT secret from {oldPath} to {newPath}");
+                    jsonRpcConfig.JwtSecretFile = newPath;
+                }
+                catch (Exception ex)
+                {
+                    if (logger.IsError) logger.Error($"Failed moving JWT secret to {newPath}.", ex);
+                    jsonRpcConfig.JwtSecretFile = oldPath;
+                }
+            }
+            else {
+                jsonRpcConfig.JwtSecretFile = newPath;
+            } 
```

Comment:
Now we can simplify it even more by setting upfront to `newPath` and falling back to `oldPath`:
```suggestion
            jsonRpcConfig.JwtSecretFile = newPath;
            // check if jwt-secret file already exists in previous default directory
            if (!File.Exists(newPath) && File.Exists(oldPath))
            {
                try
                {
                    File.Move(oldPath, newPath);
                    
                    if (logger.IsWarn) logger.Warn($"Moved JWT secret from {oldPath} to {newPath}");
                }
                catch (Exception ex)
                {
                    if (logger.IsError) logger.Error($"Failed moving JWT secret to {newPath}.", ex);
                    
                    jsonRpcConfig.JwtSecretFile = oldPath;
                }
            }
```